sonic_platform_base/sonic_xcvr/api/public/sff8472.py

```diff
@@ -247,20 +247,7 @@ def tx_disable(self, tx_disable):
         return self.xcvr_eeprom.write(consts.TX_DISABLE_SELECT_FIELD, tx_disable)
 
     def tx_disable_channel(self, channel, disable):
-        channel_state = self.get_tx_disable_channel()
-        if channel_state is None or channel_state == "N/A":
-            return False
-
-        for i in range(self.NUM_CHANNELS):
-            mask = (1 << i)
-            if not (channel & mask):
-                continue
-            if disable:
-                channel_state |= mask
-            else:
-                channel_state &= ~mask
-
-        return self.xcvr_eeprom.write(consts.TX_DISABLE_FIELD, channel_state)
+        return self.tx_disable(disable) if channel != 0 else True
```

Comment:
@prgeor channel here is the bit based mask so value of 0 means no channel should be disabled.
Since test_tx_disable_channel passes channel as 0 as well for every port, the above check is to ensure that SFF-8472 ignores such request to disable TX.